addons/mail/static/src/messaging/messaging_env.js

```diff
@@ -57,18 +51,16 @@ function addMessagingToEnv(env) {
              * applied before messaging initialization.
              */
             window.addEventListener('load', resolve);
-        }).then(async () => {
-            /**
-             * All JS resources are loaded, but not necessarily processed.
-             * We assume no messaging-related modules return any Promise,
-             * therefore they should be processed *at most* asynchronously at
-             * "Promise time".
-             */
-            await new Promise(resolve => setTimeout(resolve));
-            _addMessagingEntities(env);
         });
+        /**
+         * All JS resources are loaded, but not necessarily processed.
+         * We assume no messaging-related modules return any Promise,
+         * therefore they should be processed *at most* asynchronously at
+         * "Promise time".
+         */
+        await new Promise(resolve => setTimeout(resolve));
```

Comment:
`await new Promise(setTimeout);`

Reply:
This looks very tricky, I feel like this is asking for bugs to not have the promise constructor explicitly define its `resolve` but rely on the API of function define elsewhere.